src/main/java/duke/util/TaskList.java

```diff
@@ -0,0 +1,160 @@
+package duke.util;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * Represents an ArrayList of Task.
+ */
+public class TaskList {
+    //List that holds all the Tasks.
+    private final ArrayList<Task> list;
+    //Storage to load the list and write to list.
+    private final Storage store;
+
+    /**
+     * Constructs List that contains Tasks.
+     *
+     * @param fileName File to save and load existing TaskList from.
+     */
+    public TaskList(String fileName) {
+        this.store = new Storage(fileName);
+        this.list = store.loadFromFile();
+    }
+
+    /**
+     * Adds ToDo Task to list.
+     *
+     * @param text Description of ToDo Task.
+     */
+    public String addToDo(String text) {
+        Task newToDo = new ToDo(text);
+        list.add(newToDo);
+        store.writeToFile(list);
+        return "Got it. I've added this duke.task: \n" +
+                newToDo +
+                "Now you have " + list.size() + " tasks in the list. \n";
+    }
+
+    /**
+     * Adds Deadline Task to list.
+     *
+     * @param text Description of Deadline.
+     * @param by Deadline of Deadline Task.
+     */
+    public String addDeadline(String text, String by) {
+        try {
+            Task newDl = new Deadline(text, by);
+            list.add(newDl);
+            store.writeToFile(list);
+            return  "Got it. I've added this duke.task: \n" +
+                    newDl +
+                    "Now you have " + list.size() + " tasks in the list. \n";
+        } catch (DukeException | DateTimeParseException e) {
+            return "Please input your date and time in dd/mm/yyyy hhmm format, eg. 28/08/2021 2359";
+        }
+
+    }
+
+    /**
+     * Adds Event Task to list.
+     *
+     * @param text Description of Event Task.
+     * @param at Date time of Event.
+     */
+    public String addEvent(String text, String at) {
+        try {
+            Task newEvent = new Event(text, at);
+            list.add(newEvent);
+            store.writeToFile(list);
+            return "Got it. I've added this duke.task: \n" +
+                    newEvent +
+                    "Now you have " + list.size() + " tasks in the list. \n";
+        } catch (DukeException | DateTimeParseException e) {
+            return "Please input your date and time in dd/mm/yyyy hhmm format, eg. 28/08/2021 2359";
+        }
+    }
+
+    /**
+     * Sets Task at list index to done .
+     *
+     * @param index Position of Task to be set to done.
+     * @return
+     */
+    public String setIndexDone(int index) {// starts from 1
```

Comment:
Maybe add the comment on line 91 to increase readability?
Same for the comment on line 91.